pkg/mdns/controller.go

```diff
@@ -73,16 +76,18 @@ func (c *MicroShiftmDNSController) Run(ctx context.Context, ready chan<- struct{
 	}
 
 	c.myIPs = ips
-
 	if strings.HasSuffix(c.NodeName, server.DefaultmDNSTLD) {
-
 		klog.Infof("mDNS: Host FQDN %q will be announced via mDNS on IPs %q", c.NodeName, ips)
 		c.resolver.AddDomain(c.NodeName+".", ips)
 	}
 
 	close(ready)
 
-	go c.startRouteInformer(c.stopCh)
+	go func() {
+		if err := c.startRouteInformer(c.stopCh); err != nil {
+			close(c.stopCh)
+		}
+	}()
 
```

Comment:
Will that result in shutting down microshift? Or will it just continue to run without this controller working properly?

Reply:
I've replaced the `close()` statement with a log ling as it's closed by the deferred call in func scope.